Client/Frontend/Browser/BrowserToolbar.swift

```diff
@@ -6,28 +6,115 @@ import Foundation
 import UIKit
 import Snap
 
+@objc
+protocol BrowserToolbarProtocol {
+    weak var browserToolbarDelegate: BrowserToolbarDelegate? { get set }
+    var shareButton: UIButton { get }
+    var bookmarkButton: UIButton { get }
+    var forwardButton: UIButton { get }
+    var backButton: UIButton { get }
+
+    func updateBackStatus(canGoBack: Bool)
+    func updateFowardStatus(canGoForward: Bool)
+    func updateBookmarkStatus(isBookmarked: Bool)
+}
+
+@objc
 protocol BrowserToolbarDelegate: class {
-    func browserToolbarDidPressBack(browserToolbar: BrowserToolbar, button: UIButton)
-    func browserToolbarDidPressForward(browserToolbar: BrowserToolbar, button: UIButton)
-    func browserToolbarDidLongPressBack(browserToolbar: BrowserToolbar, button: UIButton)
-    func browserToolbarDidLongPressForward(browserToolbar: BrowserToolbar, button: UIButton)
-    func browserToolbarDidPressBookmark(browserToolbar: BrowserToolbar, button: UIButton)
-    func browserToolbarDidLongPressBookmark(browserToolbar: BrowserToolbar, button: UIButton)
-    func browserToolbarDidPressShare(browserToolbar: BrowserToolbar, button: UIButton)
+    func browserToolbarDidPressBack(browserToolbar: BrowserToolbarProtocol, button: UIButton)
+    func browserToolbarDidPressForward(browserToolbar: BrowserToolbarProtocol, button: UIButton)
+    func browserToolbarDidLongPressBack(browserToolbar: BrowserToolbarProtocol, button: UIButton)
+    func browserToolbarDidLongPressForward(browserToolbar: BrowserToolbarProtocol, button: UIButton)
+    func browserToolbarDidPressBookmark(browserToolbar: BrowserToolbarProtocol, button: UIButton)
+    func browserToolbarDidLongPressBookmark(browserToolbar: BrowserToolbarProtocol, button: UIButton)
+    func browserToolbarDidPressShare(browserToolbar: BrowserToolbarProtocol, button: UIButton)
 }
 
-private let ButtonInset = UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10)
+@objc
+public class BrowserToolbarHelper {
```

Comment:
What's the point of this helper? Between this and the protocol, it seems like you're trying to extract functionality here so we can reuse code between portrait/landscape modes. But I don't see you using either one in more than one place, so it's kind of confusing.

If you have plans to have multiple `BrowserToolbarProtocol` implementations that reuse the helper functionality, could you move these to that patch? Otherwise, what are these for?